packages/node-core/src/indexer/fetch.service.ts

```diff
@@ -196,6 +198,23 @@ export class FetchService<DS extends BaseDataSource, B extends IBlockDispatcher<
       // Update the target height, this happens here to stay in sync with the rest of indexing
       void this.storeModelProvider.metadata.set('targetHeight', latestHeight);
 
+      // If we're rewinding, we should wait until it's done
+      const multiChainStatus = this.multiChainRewindService.status;
+
+      if (RewindStatus.Rewinding === multiChainStatus) {
+        assert(this.multiChainRewindService.waitRewindHeader, 'Multi chain Rewind header is not set');
```

Comment:
I think it would be worth combining `status` and `waitRewindHeader` into a single response. It would clarify the usage and makes the multichain rewind service ensure that the header is available. 

There is no clear definition here as to how the fetch service would recover from `waitRewindHeader` being undefined.